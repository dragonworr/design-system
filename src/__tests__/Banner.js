import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import { Banner, theme } from '..'

describe('Banner', () => {
  test('renders with no props other than theme', () => {
    const json = renderer.create(<Banner theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with custom iconName and size', () => {
    const json = renderer
      .create(<Banner iconName="star" iconSize={20} theme={theme} />)
      .toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with blue bg', () => {
    const json = renderer.create(<Banner bg="blue" theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.blue)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('renders with green bg', () => {
    const json = renderer.create(<Banner bg="green" theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.green)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('renders with orange bg', () => {
    const json = renderer.create(<Banner bg="orange" theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.orange)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('renders with red bg', () => {
    const json = renderer.create(<Banner bg="red" theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.red)
    expect(json).toHaveStyleRule('color', theme.colors.white)
  })

  test('renders with lightBlue bg', () => {
    const json = renderer
      .create(<Banner bg="lightBlue" theme={theme} />)
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightBlue)
    expect(json).toHaveStyleRule('color', theme.colors.text)
  })

  test('renders with lightGreen bg', () => {
    const json = renderer
      .create(<Banner bg="lightGreen" theme={theme} />)
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightGreen)
    expect(json).toHaveStyleRule('color', theme.colors.text)
  })

  test('renders with lightOrange bg', () => {
    const json = renderer
      .create(<Banner bg="lightOrange" theme={theme} />)
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightOrange)
    expect(json).toHaveStyleRule('color', theme.colors.text)
  })

  test('renders with lightRed bg', () => {
    const json = renderer
      .create(<Banner bg="lightRed" theme={theme} />)
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.lightRed)
    expect(json).toHaveStyleRule('color', theme.colors.text)
  })

  test('renders close button if onClose func is provided', () => {
    const wrapper = shallow(<Banner onClose={() => {}} theme={theme} />)
    const closeButton = wrapper.find('CloseButton')
    expect(closeButton).toHaveLength(1)
  })

  test('does render blue left-hand icon by default', () => {
    const wrapper = shallow(<Banner bg="blue" theme={theme} />)
    const icon = wrapper.find('[name="information"]')
    expect(icon).toHaveLength(1)
  })

  test('does render green left-hand icon by default', () => {
    const wrapper = shallow(<Banner bg="green" theme={theme} />)
    const icon = wrapper.find('[name="success"]')
    expect(icon).toHaveLength(1)
  })

  test('does render orange left-hand icon by default', () => {
    const wrapper = shallow(<Banner bg="orange" theme={theme} />)
    const icon = wrapper.find('[name="attention"]')
    expect(icon).toHaveLength(1)
  })

  test('does render red left-hand icon by default', () => {
    const wrapper = shallow(<Banner bg="red" theme={theme} />)
    const icon = wrapper.find('[name="warning"]')
    expect(icon).toHaveLength(1)
  })

  test('does not render blue left-hand icon if showIcon is false', () => {
    const wrapper = shallow(<Banner bg="blue" showIcon={false} theme={theme} />)
    const icon = wrapper.find('[name="information"]')
    expect(icon).toHaveLength(0)
  })

  test('does not render green left-hand icon if showIcon is false', () => {
    const wrapper = shallow(
      <Banner bg="green" showIcon={false} theme={theme} />
    )
    const icon = wrapper.find('[name="success"]')
    expect(icon).toHaveLength(0)
  })

  test('does not render orange left-hand icon if showIcon is false', () => {
    const wrapper = shallow(
      <Banner bg="orange" showIcon={false} theme={theme} />
    )
    const icon = wrapper.find('[name="attention"]')
    expect(icon).toHaveLength(0)
  })

  test('does not render red left-hand icon if showIcon is false', () => {
    const wrapper = shallow(<Banner bg="red" showIcon={false} theme={theme} />)
    const icon = wrapper.find('[name="warning"]')
    expect(icon).toHaveLength(0)
  })

  test('only accepts preset colors', () => {
    const json = renderer.create(<Banner bg={'gray'} theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).not.toHaveStyleRule('background-color', theme.colors.gray)
  })
})
