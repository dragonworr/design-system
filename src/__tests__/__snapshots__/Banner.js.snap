// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner only accepts preset colors 1`] = `
.c1 {
  text-align: flex-start;
}

.c2 {
  width: 100%;
  text-align: left;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  font-size: 14px;
}

.c3 {
  font-size: 16px;
  margin: 0px;
}

<div
  className=""
  color={undefined}
>
  <div
    className="c0 c1"
  >
    <div
      className="c2"
      width={1}
    >
      <h5
        className="c3"
        fontSize={2}
      />
      <span
        className="c4"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with blue bg 1`] = `
.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c0 {
  background-color: #007aff;
  color: #fff;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 20 20"
      width={24}
    >
      <path
        d="M10,0C4.5,0,0,4.5,0,10s4.5,10,10,10s10-4.5,10-10S15.5,0,10,0z M11,15H9V9h2V15z M11,7H9V5h2V7
      	z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      />
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with custom iconName and size 1`] = `
.c0 {
  background-color: #0a0;
  color: #fff;
}

.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 20 19"
      width={24}
    >
      <path
        d="M10,15.3l6.2,3.7l-1.6-7L20,7.2l-7.2-0.6L10,0L7.2,6.6L0,7.2L5.5,12l-1.6,7L10,15.3z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      />
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with green bg 1`] = `
.c0 {
  background-color: #0a0;
  color: #fff;
}

.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 20 20"
      width={24}
    >
      <path
        d="M10,0C4.5,0,0,4.5,0,10s4.5,10,10,10s10-4.5,10-10S15.5,0,10,0z M8,15l-5-5l1.4-1.4L8,12.2
      	l7.6-7.6L17,6L8,15z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      />
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with lightBlue bg 1`] = `
.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c0 {
  background-color: #cdf;
  color: #001833;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="text"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 20 20"
      width={24}
    >
      <path
        d="M10,0C4.5,0,0,4.5,0,10s4.5,10,10,10s10-4.5,10-10S15.5,0,10,0z M11,15H9V9h2V15z M11,7H9V5h2V7
      	z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      />
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with lightGreen bg 1`] = `
.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c0 {
  background-color: #cec;
  color: #001833;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="text"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 20 20"
      width={24}
    >
      <path
        d="M10,0C4.5,0,0,4.5,0,10s4.5,10,10,10s10-4.5,10-10S15.5,0,10,0z M8,15l-5-5l1.4-1.4L8,12.2
      	l7.6-7.6L17,6L8,15z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      />
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with lightOrange bg 1`] = `
.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c0 {
  background-color: #fec;
  color: #001833;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="text"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 22 19"
      width={24}
    >
      <path
        d="M0,19h22L11,0L0,19z M12,16h-2v-2h2V16z M12,12h-2V8h2V12z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      />
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with lightRed bg 1`] = `
.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c0 {
  background-color: #fcc;
  color: #001833;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="text"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 20 20"
      width={24}
    >
      <path
        d="M10,0C4.5,0,0,4.5,0,10s4.5,10,10,10s10-4.5,10-10S15.5,0,10,0z M11,11H9V5h2V11z M11,15H9v-2h2
      	V15z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      />
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with no props other than theme 1`] = `
.c0 {
  background-color: #0a0;
  color: #fff;
}

.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 20 20"
      width={24}
    >
      <path
        d="M10,0C4.5,0,0,4.5,0,10s4.5,10,10,10s10-4.5,10-10S15.5,0,10,0z M8,15l-5-5l1.4-1.4L8,12.2
      	l7.6-7.6L17,6L8,15z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      />
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with orange bg 1`] = `
.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c0 {
  background-color: #f90;
  color: #fff;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 22 19"
      width={24}
    >
      <path
        d="M0,19h22L11,0L0,19z M12,16h-2v-2h2V16z M12,12h-2V8h2V12z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      />
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;

exports[`Banner renders with red bg 1`] = `
.c2 {
  text-align: flex-start;
}

.c4 {
  width: 100%;
  text-align: left;
}

.c0 {
  background-color: #c00;
  color: #fff;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 14px;
}

.c3 {
  margin-right: 32px;
  margin-top: -2px;
}

.c5 {
  font-size: 16px;
  margin: 0px;
}

<div
  className="c0"
  color="white"
>
  <div
    className="c1 c2"
  >
    <svg
      className="c3"
      fill="currentcolor"
      height={24}
      mr={3}
      mt="-2px"
      viewBox="0 0 20 20"
      width={24}
    >
      <path
        d="M10,0C4.5,0,0,4.5,0,10s4.5,10,10,10s10-4.5,10-10S15.5,0,10,0z M11,11H9V5h2V11z M11,15H9v-2h2
      	V15z"
      />
    </svg>
    <div
      className="c4"
      width={1}
    >
      <h5
        className="c5"
        fontSize={2}
      />
      <span
        className="c6"
        fontSize={1}
      />
    </div>
  </div>
</div>
`;
